browser-test/src/admin/admin_program_visibility.test.ts

```diff
@@ -272,4 +273,47 @@ test.describe('Validate program visibility is correct for applicants and TIs', (
       'program-visibility-changes-all-ti-can-see-program',
     )
   })
+  test('create a program with disabled visibility, verify it is hidden from applicants and TIs', async () => {
+    const {page, tiDashboard, adminPrograms} = ctx
+
+    await enableFeatureFlag(page, 'disabled_visibility_condition_enabled')
+    await loginAsAdmin(page)
+
+    const programName = 'Disabled program'
+    const programDescription = 'Description'
+    await adminPrograms.addProgram(
+      programName,
+      programDescription,
+      'https://usa.gov',
+      ProgramVisibility.DISABLED,
+    )
+    await adminPrograms.publishAllDrafts()
+
+    // Login as applicant, verify program is hidden
```

Comment:
suggestion (non-blocking): use [`test.step`](https://playwright.dev/docs/api/class-test#test-step) to break up these steps into testable chunks of code. If any of them fail, it will be more clear which step they failed in. Here's an [example I recently added](https://github.com/civiform/civiform/blob/a03f3c9ccef88d279f76abaf856bc63198ace892/browser-test/src/trusted_intermediary.test.ts#L982).

Reply:
that totally makes sense since it makes it easier to debug! good suggestion, updated the test.